core/src/main/scala/com/spotify/featran/transformers/OneHotEncoder.scala

```diff
@@ -53,39 +62,62 @@ private class OneHotEncoder(name: String) extends BaseHotEncoder[String](name) {
           fb.add(name + '_' + k, 1.0)
           fb.skip(math.max(0, c.size - v - 1))
         case None =>
-          fb.skip(c.size)
+          addMissingItem(c, fb)
           fb.reject(this, FeatureRejection.Unseen(Set(k)))
       }
-      case None => fb.skip(c.size)
+      case None => addMissingItem(c, fb)
     }
   }
 }
 
-private abstract class BaseHotEncoder[A](name: String)
+object MissingValue {
```

Comment:
Added in order to be able to import into tests so tests won't have to be updated if `missingValueToken` changes.